src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala

```diff
@@ -354,6 +373,17 @@ private class TaskLauncherActor(
     promise.trySuccess(MatchedInstanceOps(offer.getId, Seq(InstanceOpWithSource(myselfAsLaunchSource, instanceOp))))
   }
 
+  private[this] def scheduleTaskOpTimeout(
+    context: ActorContext,
+    instanceOp: InstanceOp): Cancellable =
+    {
+      import context.dispatcher
+      val message: InstanceOpSourceDelegate.InstanceOpRejected = InstanceOpSourceDelegate.InstanceOpRejected(
+        instanceOp, TaskLauncherActor.OfferOperationRejectedTimeoutReason
+      )
+      context.system.scheduler.scheduleOnce(config.taskOpNotificationTimeout().milliseconds, self, message)
```

Comment:
I think we should cancel this when actor stops... Previously we were holding map of Cancellable items and cancelling them in postStop. I think we should do something similar here.

Reply:
I thought about this. IMHO keep track of cancellable items just makes the implementation too complex. I can add it again but would not.

Reply:
The first way is to set up periodic message scheduling in the constructor of the actor, and cancel that scheduled sending in postStop or else we might have multiple registered message sends to the same actor.

Reply:
@jeschkies if we do not cancel then we will continue to send messages after the actor dies, we'll get a dead-letter.

```
/Users/tim/src/oss/akka/akka-actor/src/main/scala/akka/actor/Scheduler.scala

  42 |    */
  43 |   final def schedule(
  44 |     initialDelay: FiniteDuration,
  45 |     interval:     FiniteDuration,
  46 |     receiver:     ActorRef,
  47 |     message:      Any)(implicit
  48 |     executor: ExecutionContext,
  49 |                        sender: ActorRef = Actor.noSender): Cancellable =
  50 |     schedule(initialDelay, interval, new Runnable {
  51 |       def run = {
  52 |         receiver ! message
  53 |         if (receiver.isTerminated)
  54 |           throw new SchedulerException("timer active for terminated actor")
  55 |       }
  56 |     })
  57 | 
```

Reply:
Ok, you convinced me.